### PR TITLE
Add a simple motiviation section to the README

### PR DESCRIPTION
**Pull request by kkamperschroer**
The intent of adding this section is to make it more clear that React Navigation is intended to replace NavigationExperimental and other navigation libraries. This hopefully will prevent confusion that this is just yet-another-navigation-library.

Some of the text was grabbed from the [introduction](https://reactnavigation.org/blog/2017/1/Introducing-React-Navigation) blog post, and some from the referenced issues.

Given this is a documentation change, no test-plan included.

Ref #28 and #33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,19 @@
...
+
+React Navigation is born from the React Native community's need for an
+extensible yet easy-to-use navigation solution. It replaces and improves
+upon several navigation libraries in the ecosystem, including Ex-Navigation,
+React Native's Navigator and NavigationExperimental components. React
+Navigation can also be used across React and React Native projects allowing
+for a higher degree of shared code.
+
+Once stable, NavigationExperimental will be depracated in favor of React
+Navigation. React Navigation is a is a collaboration between people from
```
Comment (knowbody) [inline README.md]: typo: `is a is a`

Comment (kkamperschroer) [inline README.md]: Good catch. Just pushed an update.

**Comment by satya164** [comment]
cc @mkonicek

**Comment by satya164** [comment]
Should we merge this @mkonicek ?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,19 @@
...
+React Navigation is born from the React Native community's need for an
+extensible yet easy-to-use navigation solution. It replaces and improves
+upon several navigation libraries in the ecosystem, including Ex-Navigation,
+React Native's Navigator and NavigationExperimental components. React
+Navigation can also be used across React and React Native projects allowing
+for a higher degree of shared code.
+
+Once stable, NavigationExperimental will be depracated in favor of React
+Navigation. React Navigation is a collaboration between people from
+Facebook, Exponent and the React community at large.
```
Comment (grabbou) [inline README.md]: Shameless self-promotion, since me and @satya164 are working for Callstack, maybe we can list us here as well? :)

**Comment by grabbou** [comment]
@satya164 I think it's good to go, just added a question whether it's good to list Callstack as well. Will try to get some traction from Martin via Slack, as this is minor :)

File: README.md (hunk 2 above)
Comment (mkonicek) [inline README.md]: Sure!

**Comment by mkonicek** [comment]
lgtm

**Pull request merged**